In [5]:
import os
from pypdf import PdfReader

# 1. Path to your manual (using raw string r"..." to avoid path errors)
manual_path = r"E:\Applied AI - iti\Project\data\manual\Toyota-Corolla_2026_EN__b4f47a2baa.pdf"  

# 2. Read the PDF
reader = PdfReader(manual_path)
total_pages = len(reader.pages)

print(f"Total pages in manual: {total_pages}")

# 3. Extract text from the first few pages to verify readability
sample_text = ""
for i in range(min(5, total_pages)):
    sample_text += f"--- PAGE {i+1} ---\n" + reader.pages[i].extract_text()

print(sample_text[:1000])  # Display the first 1000 characters

Total pages in manual: 497
--- PAGE 1 ---
C
1
M
E
J
S
K
L
U
D
R
2
3
4
5
6
7
8
9
0--- PAGE 2 ---
--- PAGE 3 ---
•• PROTECTED 関係者外秘 PROTÉGÉ 
NOTICE TO QUEBEC CONSUMERS: 
Toyota Canada Inc. is pleased to offer a 36 month / 60, 000 k m 
(whichever occurs first) limited ne w -vehicle warranty on your 
vehicle. This limited warranty covers repairs on any part of the 
vehicle supplied by Toyota Canada that is defective in material 
or workmanship, subject to certain exceptions that are stated 
in your warranty manual. For more details of the warranty, 
check the Owner’s Manual Supplement included with your 
vehicle, or visit toyota.ca.  
Quebec’s Consumer Protection Act requires Toyota Canada to 
disclose whether we guarantee the availability of replacement 
parts, repair services, and information necessary to maintain 
or repair the vehicle or its components. While Toyota Canada is 
pleased to honour the terms of our limited warranty, please 
note that Toyota in no way guarantees the availabi

### 2.1 Inspection Summary
- **Total Documents:** 1 PDF (`Ford_F150_Manual.pdf` / `Toyota_Corolla_Manual.pdf`)
- **Total Pages:** 497 pages
- **Parsing Status:** Text extracted successfully via `pypdf`. Pages 1–2 contain cover artwork/codes, while standard text begins on Page 3. No OCR needed.

In [6]:
from pypdf import PdfReader

# 1. Load all text from the manual
reader = PdfReader(manual_path)
documents = []

for idx, page in enumerate(reader.pages):
    text = page.extract_text()
    if text and text.strip():  # Filter out empty pages
        documents.append({"page": idx + 1, "text": text})

print(f"Extracted non-empty pages: {len(documents)}")


# 2. Chunking Function (Fixed-size with overlap)
def chunk_text(docs, chunk_size=1000, overlap=100):
    chunks = []
    for doc in docs:
        page_num = doc["page"]
        text = doc["text"]

        start = 0
        while start < len(text):
            end = start + chunk_size
            chunk_str = text[start:end]

            chunks.append(
                {
                    "text": chunk_str,
                    "metadata": {
                        "source": f"Manual - Page {page_num}",
                        "page": page_num,
                    },
                }
            )

            start += chunk_size - overlap
    return chunks


# 3. Create chunks
all_chunks = chunk_text(documents, chunk_size=1000, overlap=100)
print(f"Total Chunks Created: {len(all_chunks)}")
print("\nSample Chunk 1:\n", all_chunks[50]["text"][:300])

Extracted non-empty pages: 495
Total Chunks Created: 1088

Sample Chunk 1:
 23
1
1For safety and security
For safety and security
.
1-1. For safe use
Before driving................. 24
For safe driving ..............25
Seat belts .......................27
SRS airbags................... 31
Front passenger occupant 
classification system ....41
Exhaust gas precautions
.......


### Chunking Justification
- **Chunk Size (1000 chars):** Captures complete safety instructions, warning descriptions, and troubleshooting steps without splitting critical sentence context.
- **Overlap (100 chars):** Ensures continuity between adjacent chunks so no key details are lost at boundaries.